### PR TITLE
http: feature/xml systemout

### PR DESCRIPTION
**Pull request by ivan-velasco**
http: addition of 'systemout' to the XML report for enhanced debugging capabilities, eliminating the need to solely rely on 'INFO' messages or enabling verbose mode after a complete run. With this enhancement, 'systemout' will be included in the XML output specifically when errors occur, improving the troubleshooting and diagnostic process.

### CODE REVIEW

**Review by alexhaywardsocotra — APPROVED** [review]

**Review by yesnault — APPROVED** [review]
Hi, thank you for this proposal. I'm not sure about the performance impact with big payload, but yes, this could be useful. LGTM.

**Comment by yesnault** [comment]
After re-reading, I think that the body will be present twice in the JSON output. Maybe we can avoid that with the json annotation to '-'.

**Review by yesnault — CHANGES_REQUESTED** [review]

File: executors/http/http.go [hunk 1]
```diff
@@ -68,6 +68,7 @@ type Result struct {
 	BodyJSON    interface{} `json:"bodyjson,omitempty" yaml:"bodyjson,omitempty"`
 	Headers     Headers     `json:"headers,omitempty" yaml:"headers,omitempty"`
 	Err         string      `json:"err,omitempty" yaml:"err,omitempty"`
+	Systemout   string      `json:"systemout,omitempty" yaml:"systemout,omitempty"`
```
[the remark this answers is not in the document]
Comment (yesnault) [reply executors/http/http.go:71]: ok with that @ivan-velasco  ?

[the remark this answers is not in the document]
Comment (ivan-velasco) [reply executors/http/http.go:71]: Going to test this change out locally today @yesnault

[the remark this answers is not in the document]
Comment (ivan-velasco) [reply executors/http/http.go:71]: @yesnault tested this finally today and it causes the xml report to exclude this information
When the property is set `Systemout   string      json:"-" yaml:"-"`
```xml
<system-out><![CDATA[
```
json report
```json
"systemout": "\n",
```

When the property is set `Systemout   string      json:"systemout,omitempty" yaml:"systemout,omitempty"`, then xml report shows up properly
```xml
<system-out><![CDATA[
PATCH-----http://
```
json report
```json
"result.systemout": "PATCH-----http://
"systemout": "PATCH-----http://
```

[the remark this answers is not in the document]
Comment (ivan-velasco) [reply executors/http/http.go:71]: @yesnault Any thoughts on this? Has been very helpful in identifying errors quickly.

**Review by yesnault — APPROVED** [review]

**Pull request merged**